runtime.md

```diff
@@ -64,34 +65,60 @@ Unless otherwise stated, generating an error MUST leave the state of the environ
 `state <container-id>`
 
 This operation MUST generate an error if it is not provided the ID of a container.
+Attempting to query a container that does not exist MUST generate an error.
 This operation MUST return the state of a container as specified in the [State](#state) section.
-In particular, the state MUST be serialized as JSON.
+The state MUST be serialized as JSON.
 
 
-### Start
+### Create
 
-`start <container-id> <path-to-bundle>`
+`create <container-id> <path-to-bundle>`
 
 This operation MUST generate an error if it is not provided a path to the bundle and the container ID to associate with the container.
-If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error.
-Using the data in `config.json`, that are in the bundle's directory, this operation MUST create a new container.
-This includes creating the relevant namespaces, resource limits, etc and configuring the appropriate capabilities for the container.
-A new process within the scope of the container MUST be created as specified by the `config.json` file otherwise an error MUST be generated.
+If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error and a new container MUST not be created.
+Using the data in [`config.json`](config.md), this operation MUST create a new container.
+This includes creating, or entering, the namespaces specified in the [`config.json`](config.md), resource limits, etc. and configuring the appropriate capabilities for the container.
+If the `config.json` specifies that a PID namespace is to be created then one MUST be created, but the user-specified code within that namespace MUST NOT be run at this time.
+In some implementations this means that a temporary process is created in the PID namespace but it pauses until the `start` operation is invoked before replacing the process with the user-specified code.
 
-Attempting to start an already running container MUST have no effect on the container and MUST generate an error.
+### Start
+`start <container-id>`
 
-### Stop
+This operation MUST generate an error if it is not provided the container ID.
+Attempting to start a container that does not exist MUST generate an error.
+This operation MUST run the user-specified code as specified by [`process`](config.md#process-configuration) otherwise an error MUST be generated and no process MUST be started.
+Note that if the [`process`](config.md#process-configuration) was modified after the `create` operation was invoked then that change MUST be used by the runtime.
+No other changes to the [`config.json`](config.md) MUST be used by the runtime.
```

Comment:
These two lines landed in the 1aae8aa0 → 8a34a63 reroll, and I'd rather rephrase the last three lines to something like:

    This operation MUST run the user-specified code as specified by [`process`](config.md#process-configuration) in the configuration file at start-time (the values may have changed since [create](#create)-time).
    If the runtime fails to run code as specified, an error MUST be generated.
    The runtime MUST ignore `config.json` properties outside of `process`.

Motivation for those changes:

* Push the “may have been modified” warning to “as specified… at start-time”, to keep the rules about which configuration to use in a single line.
* Reword “No other changes… MUST be used” to “MUST ignore” to avoid the space between the negation and the MUST.
* Drop “no process MUST be started”, because the container process was already started at create-time.  It might be possible to have a rule like “start cannot make changes if it would fail to complete”, but that sounds tricky to support.  Even a more restricted rule like “if start fails, you can try to call it again” would be tricky to require (what if the reason it failed was that something had killed the socket it used to talk to the dummy init?).  It seems better to just drop the requirement and leave it to runtimes to do their best to handle errors appropriately.